### PR TITLE
docs: align supported engines with info from main repo

### PR DESCRIPTION
**Pull request by ephys**
The information in /releases wasn't in line with the `ENGINES.md` document on our main repo (https://github.com/sequelize/sequelize/blob/main/ENGINE.md)

This PR should fix that.

### CODE REVIEW

File: src/pages/releases.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ MariaDB requires the use of the [mariadb][mariadb-npm] npm package.
 | Sequelize   | [MariaDB][mariadb] | [mariadb (npm)][mariadb-npm] |
 |-------------|--------------------|------------------------------|
 | 7 (alpha)   | >=10.3             | >= 3.0.0 [^⚠️2]              |
-| 6 (current) | >=10.3             | ^2.3.3                       |
+| 6 (current) | >=10.1.44          | ^2.3.3                       |
```
Comment (WikiRik) [inline src/pages/releases.md:52]: We're not testing on 10.1 right now, but would be good to add that

**Review by WikiRik — APPROVED** [review]

File: src/pages/releases.md (hunk 1 above)
Comment (ephys) [reply src/pages/releases.md:52]: We're not, but we should look into that. I think we shouldn't remove support for a database version without doing a major release

Comment (WikiRik) [reply src/pages/releases.md:52]: I'll look into that this week

Comment (ephys) [reply src/pages/releases.md:52]: Well, looks like this one needs to be reverted

**Review by WikiRik — APPROVED** [review]

**Comment by ephys** [comment]
Thanks :)

**Pull request merged**